tests/api/test_product.py

```diff
@@ -3961,3 +3961,184 @@ def test_update_or_create_variant_stocks_empty_stocks_data(variant, warehouses):
     stock = variant.stocks.first()
     assert stock.warehouse == warehouses[0]
     assert stock.quantity == 5
+
+
+# Because we use Scalars for Weight this test query tests only a scenario when weight
+# value is passed by a variable
+MUTATION_CREATE_PRODUCT_WITH_WEIGHT_GQL_VARIABLE = """
+mutation createProduct(
+        $productType: ID!,
+        $category: ID!
+        $name: String!,
+        $sku: String,
+        $basePrice: Decimal!
+        $weight: WeightScalar
+        $trackInventory: Boolean)
+    {
+        productCreate(
+            input: {
+                category: $category,
+                productType: $productType,
+                name: $name,
+                sku: $sku,
+                trackInventory: $trackInventory,
+                basePrice: $basePrice,
+                weight: $weight
+            })
+        {
+            product {
+                id
+                name
+                variants{
+                    id
+                    sku
+                    trackInventory
+                    quantity
+                }
+                weight{
+                    value
+                    unit
+                }
+            }
+            productErrors {
+                message
+                field
+                code
+            }
+        }
+    }
+    """
+
+# Because we use Scalars for Weight this test query tests only a scenario when weight
+# value is passed by directly in input
+MUTATION_CREATE_PRODUCT_WITH_WEIGHT_GQL_INPUT = """
+mutation createProduct(
+        $productType: ID!,
+        $category: ID!
+        $name: String!,
+        $sku: String,
+        $basePrice: Decimal!
+        $trackInventory: Boolean)
+    {{
```

Comment:
Duble `{`

Reply:
This is needed if You want to use '{ }' in string with '.format()'.
We need to use '.format()' because we want to check if passing data directly in input weight field works.

Reply:
Use `%` instead of format like in `test_query_countries_with_translation`. In this solution, we don't have doubled `{`.

Reply:
`%` smells like 90ties. Is that a real problem? It is used only in this test and You can see for the first sight that it is different than the rest of defined queries so probably you shouldn't use it in any other test. And maybe we should move it inside the TC and use `f-strings` instead?

Reply:
It's no longer Python 2.7, let's use f-strings where we can.